Add ETCD_IMAGE_TAG & ETCD_IMAGE_URL options for etcd-member.service. Use etcd3 gateway instead of etcd2 proxy.

puneetkatyal opened the pull request:
Add ETCD_IMAGE_TAG & ETCD_IMAGE_URL options for etcd-member.service
    - etcd-member.service is the recommended way to setup etcd (v3 onwards)
      on CoreOS.
    - https://coreos.com/etcd/docs/latest/getting-started-with-etcd.html#setting-up-etcd
       "Container Linux's etcd-member.service systemd unit knows how to fetch
       and run the current etcd v3.x container image, providing etcd v3 without
       requiring the binary to be present in every default OS installation."
    - This change updates 10-peers.conf.j2 to specify etcd version and url using
      ETCD_IMAGE_TAG & ETCD_IMAGE_URL environment variables
      (See /usr/lib/coreos/etcd-wrapper for more details)

Use etcd3 gateway instead of etcd2 proxy    
    - "proxy" supports v2 API only.
      (https://coreos.com/etcd/docs/latest/op-guide/configuration.html#proxy-flags)
    - etcd3 gateway is same as etcd2 proxy
      (https://github.com/coreos/etcd/issues/7439)
    - Updating 10-peers.conf.j2 as per
      https://github.com/coreos/matchbox/blob/master/examples/ignition/etcd3-gateway.yaml

puneetkatyal committed:
Add ETCD_IMAGE_TAG & ETCD_IMAGE_URL options for etcd-member.service

- etcd-member.service is the recommended way to setup etcd (v3 onwards)
  on CoreOS.
- https://coreos.com/etcd/docs/latest/getting-started-with-etcd.html#setting-up-etcd
   "Container Linux's etcd-member.service systemd unit knows how to fetch
   and run the current etcd v3.x container image, providing etcd v3 without
   requiring the binary to be present in every default OS installation."
- This change updates 10-peers.conf.j2 to specify etcd version and url using
  ETCD_IMAGE_TAG & ETCD_IMAGE_URL environment variables
  (See /usr/lib/coreos/etcd-wrapper for more details)

imkin commented on defaults/main.yml:
```diff
@@ -9,4 +9,6 @@ etcd_ca_depot: ""
\ No newline at end of file
+etcd_core_group: ""
+
+etcd_image_tag: "v3.1.0"
```
Where is the etcd image?

puneetkatyal replied:
https://coreos.com/etcd/docs/latest/getting-started-with-etcd.html#setting-up-etcd
> Container Linux's etcd-member.service systemd unit knows how to fetch and run the current etcd v3.x container image, providing etcd v3 without requiring the binary to be present in every default OS installation.

imkin replied:
I meant the location of etcd image. If it is pulled from coreos the setup might have issues like we have with existing images taking a lot of time to download.

puneetkatyal replied:
Its's downloaded from quoy.io by default. I've added an option to update ETCD_IMAGE_URL to a different source.

imkin commented:
Can you add description to this. It is usually pulled from the commit message. 

How will this branching work between current 1.4 and 1.7 features.

puneetkatyal committed:
Use etcd3 gateway instead of etcd2 proxy

- "proxy" supports v2 API only.
  (https://coreos.com/etcd/docs/latest/op-guide/configuration.html#proxy-flags)
- etcd3 gateway is same as etcd2 proxy
  (https://github.com/coreos/etcd/issues/7439)
- Updating 10-peers.conf.j2 as per
  https://github.com/coreos/matchbox/blob/master/examples/ignition/etcd3-gateway.yaml

puneetkatyal commented:
Updated commit description and add change to fix proxy mode (a.k.a. 'gateway') for etcd3
